packages/@expo/cli/src/start/doctor/dependencies/__tests__/validateDependenciesVersions-test.ts

```diff
@@ -192,6 +192,50 @@ describe(validateDependenciesVersionsAsync, () => {
     expect(Log.warn).not.toHaveBeenCalledWith(expect.stringContaining('expo-splash-screen'));
   });
 
+  it('supports npm package args for excluded packages', async () => {
+    jest.mocked(Log.warn).mockReset();
+    vol.fromJSON(
+      {
+        'node_modules/expo/package.json': JSON.stringify({
+          version: '41.0.0',
+        }),
+        'node_modules/expo-splash-screen/package.json': JSON.stringify({
+          version: '0.2.3',
+        }),
+        'node_modules/expo-updates/package.json': JSON.stringify({
+          version: '1.3.4',
+        }),
+        'node_modules/firebase/package.json': JSON.stringify({
+          version: '10.0.0',
+        }),
+      },
+      projectRoot
+    );
+    const exp = {
+      sdkVersion: '41.0.0',
+    };
+    const pkg = {
+      dependencies: {
+        'expo-splash-screen': '~0.2.3',
+        'expo-updates': '~1.3.4',
+        firebase: '~10.0.0',
+      },
+      // "Don't validate this for me for me as long as you plan to recommend @~0.2.3 - I don't want that version"
+      expo: { install: { exclude: ['expo-splash-screen@~1.2.3', 'firebase@9.1.0'] } },
```

Comment:
Tried this locally and changed `expo-splash-screen@~1.2.3` -> `expo-splash-screen@1.2.3` and it's prompting me to upgrade. Is that correct? In this case:
- node_modules: `0.2.3`
- dependencies: `~0.2.3`
- exclude: `1.2.3`
- getCombinedKnownVersionsAsync: `~1.2.3`

I would expect it _not_ to prompt since `~1.2.3` is included in `1.2.3`

Reply:
yeah i think it's not necessarily intuitive. if you provide an exact version to exclude, it only excludes that version if that is the exact version specified in the combined known versions. if you specify a range, then it is excluded if the given range is a superset of the verified range. we could change it so that if you specify a version then it is ignore if it is included in the range

Reply:
I see! That could definitely get confusing. I guess looking at this from the point of view of the user, you want to be prompted again if the recommended version has changed from that which you'd previously opted out of.
Would it be too naiive to do an _exact_ match instead?

So let's say Doctor is recommending `~1.2.3` so if you have exactly `~1.2.3` excluded, then it will not prompt you, but if Doctor had `~1.2.4` and you had `~1.2.3` it would prompt you so you'd have to decide whether to continue to ignore it (in which case you'd update to `~1.2.4`) or not.

Pros: you don't miss any releases by misconfiguring the ranges you want to opt out of
Cons: it might be conversely confusing to define a range but it being an exact match (though at least it'd be easier to explain)

Reply:
the problem with following the exact example you describe is that it doesn't comply with what the semver operators actually mean. if you've specified `~1.2.3` it effectively means "i'm ok with any 1.2.x version where the patch version is greater or equal to 3" - you wouldn't be warned for 1.2.4.

we could improve this by not using ranges in bundled native modules - so our recommended version would be `1.2.3` and then we could specify `package@<=1.2.3` in the exclude list and that would let you know when 0.74 is released. this would work without any code changes in this pr, we'd just have to change how we reference bundled native modules. i'd actually be OK with this, we typically want to recommend using the latest version of a package and warn otherwise

Reply:
I like that, that definitely seems the least confusing. Documentation will be needed regardless, but comparison operators would definitely communicate the intent better!